### ISSUE TITLE
Client does not receive message through channel he just subscribed to.

### ISSUE THREAD

**Issue by trafium** [anycable/anycable#93]
**Ruby version:** 2.6.3

**`anycable` gem version:** 0.6.3
**`anycable-rails` gem version:** 0.6.4

**`grpc` gem version:** 1.24.0

### What did you do?
In `subscribed` method of my channel I assign stream for connection and immediately broadcast a message to that stream.

Channel:
```ruby
class TestChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'test_room'
    ActionCable.server.broadcast('test_room', 'Hello world!')
  end
end
```

Frontend subscription: 
```js
App.cable.subscriptions.create({
  channel: 'TestChannel'
}, {
  received (data) {
    alert(data);
  }
});
```

### What did you expect to happen?
Subscribed browser tab will immediately receive a `'Hello world!'` message.
It worked like that with ActionCable

### What actually happened?
Subscribed browser tab does not receive any messages (not until another tab subscribes, but that is expected).

**Comment by palkan** [comment]
This is an expected non-deterministic behavior.

tl;dr Broadcast and subscription are happening asynchronously and independently. You can not guarantee that the broadcast will send a message to the client after its subscription is registered.

That's true for both Action Cable and AnyCable, though the details are different:
- Action Cable registers the subscription in [the pubsub event loop](https://github.com/rails/rails/blob/6a27a88b9cd05e0a828b812208fb0a8c46242eaf/actioncable/lib/action_cable/channel/streams.rb#L87-L92) (i.e., concurrently to the main execution flow); `broadcast` [publish](https://github.com/rails/rails/blob/6a27a88b9cd05e0a828b812208fb0a8c46242eaf/actioncable/lib/action_cable/subscription_adapter/redis.rb#L29) a message to Redis pub/sub, which is broadcasted to the client [as soon as received by the listener](https://github.com/rails/rails/blob/6a27a88b9cd05e0a828b812208fb0a8c46242eaf/actioncable/lib/action_cable/subscription_adapter/redis.rb#L103-L105), i.e. here we have an nondeterministic relied on Redis.
- AnyCable registers the subscription as soon as it [finishes processing the corresponding RPC call](https://github.com/anycable/anycable-go/blob/a6b839b622844432a21dabf03342b539ef7f096e/node/node.go#L330-L334); the `broadcast` flow is the same as in Action Cable.

**Comment by trafium** [comment]
@palkan 
Thank you for your response. 

If I understood correctly, overall behaviour is indeed different from vanilla AC and in some cases could be breaking to the application. If so it would be nice to have it documented somewhere.
